automake: add subdir-objects option

Our version of automake warns if this option is enabled and source files
in subdirectories are used.

It doesn't really seems to have a noticable effect on the build, but it
does remove a lot of annoying warnings.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

diff -Nrup vlc-2.1.2.orig/configure.ac vlc-2.1.2/configure.ac
--- vlc-2.1.2.orig/configure.ac	2014-02-27 00:22:19.512944952 +0100
+++ vlc-2.1.2/configure.ac	2014-02-27 00:24:21.360940651 +0100
@@ -24,7 +24,7 @@ AC_CANONICAL_BUILD
 AC_CANONICAL_HOST
 AC_PRESERVE_HELP_ORDER
 
-AM_INIT_AUTOMAKE(tar-ustar color-tests foreign)
+AM_INIT_AUTOMAKE(tar-ustar color-tests foreign subdir-objects)
 AC_CONFIG_HEADERS([config.h])
 
 # Disable with "./configure --disable-silent-rules" or "make V=1"
